### PR TITLE
more helper functions for `Partial.evaluate`

### PR DESCRIPTION
**Pull request by cdisselkoen**
*Issue #, if available:*

As suggested in comments on #281 

*Description of changes:*

Factor out more of `Partial.evaluate` into helper functions

Open to suggestions for better names.

### CODE REVIEW

**Review by khieta — APPROVED** [review]

File: cedar-lean/Cedar/Partial/Evaluator.lean [hunk 1]
```diff
@@ -67,34 +83,61 @@ def hasAttr (v : Spec.Value) (a : Attr) (es : Partial.Entities) : Result Spec.Va
...
+    | .value v₁ => Partial.getAttr v₁ a es
+    | .residual r => .ok (.residual (Partial.Expr.getAttr r a)) -- TODO more precise: pv will be a .residual if it contains any unknowns, but we might have a concrete value for the particular attr we care about
+
 /-- Analogous to Spec.bindAttr but for partial values -/
 def bindAttr (a : Attr) (res : Result Partial.Value) : Result (Attr × Partial.Value) := do
   let v ← res
   .ok (a, v)
 
+/-- Partial-evaluate a Var. No analogue in Spec.Evaluator; Spec.evaluate handles the `.var` case inline -/
+def evaluateVar (v : Var) (req : Partial.Request) : Result Partial.Value :=
```
Comment (khieta) [inline cedar-lean/Cedar/Partial/Evaluator.lean:111]: One more nit: I think it'd be nice and consistent if all the helper functions were called `evaluate*`.

File: cedar-lean/Cedar/Partial/Evaluator.lean [hunk 2]
```diff
@@ -67,34 +83,61 @@ def hasAttr (v : Spec.Value) (a : Attr) (es : Partial.Entities) : Result Spec.Va
...
+    let val ← Partial.hasAttr v₁ a es
+    .ok (.value val)
+  | .residual r => .ok (.residual (Partial.Expr.hasAttr r a)) -- TODO more precise: even though pv is a residual we may know concretely whether it contains the particular attr we care about
+
 /-- Analogous to Spec.getAttr but for partial entities -/
 def getAttr (v : Spec.Value) (a : Attr) (es : Partial.Entities) : Result Partial.Value := do
   let r ← Partial.attrsOf v es.attrs
   r.findOrErr a attrDoesNotExist
 
+/-- Analogous to Spec.getAttr but for partial value and partial entities -/
```
[the remark this answers is not in the document]
Comment (cdisselkoen) [reply cedar-lean/Cedar/Partial/Evaluator.lean:99]: I agree, but they are both analogous.  Do you have a suggestion for which one is "more analogous"? or how to clarify this in the comments? or what names we should use for the two of these?

[the remark this answers is not in the document]
Comment (cdisselkoen) [reply cedar-lean/Cedar/Partial/Evaluator.lean:99]: I think the ones without the `p` subscript are "more analogous", and the ones with the "p" subscript could perhaps be renamed with the `evaluate*` convention you suggest below

[the remark this answers is not in the document]
Comment (khieta) [reply cedar-lean/Cedar/Partial/Evaluator.lean:99]: Yeah that'd be my suggestion -- leave the comment on the ones without p and rename the ones with p.

File: cedar-lean/Cedar/Partial/Evaluator.lean (hunk 1 above)
Comment (cdisselkoen) [reply cedar-lean/Cedar/Partial/Evaluator.lean:123]: Renamed the functions with `p` subscripts to `evaluate*` (and also renamed `call` to `evaluateCall`) in the latest revision.

**Review by khieta — APPROVED** [review]

**Review by emina — APPROVED** [review]

**Pull request merged**